### PR TITLE
add ConnectionRefusedError raising

### PR DESCRIPTION
**Pull request by basfroman**

### CODE REVIEW

File: bittensor/core/subtensor.py [hunk 1]
```diff
@@ -186,6 +186,7 @@ def __init__(
 
         self.log_verbose = log_verbose
         self._connection_timeout = connection_timeout
+        self.substrate: "SubstrateInterface" = None
```
Comment (thewhaleking) [inline bittensor/core/subtensor.py:189]:
```suggestion
        self.substrate: Optional["SubstrateInterface"] = None
```

File: bittensor/core/subtensor.py [hunk 2]
```diff
@@ -223,14 +225,15 @@ def _get_substrate(self):
...
+        except ConnectionRefusedError as error:
             logging.error(
                 f"Could not connect to {self.network} network with {self.chain_endpoint} chain endpoint.",
             )
             logging.info(
                 "You can check if you have connectivity by running this command: nc -vz localhost "
-                f"{self.chain_endpoint.split(':')[2]}"
+                f"{self.chain_endpoint}"
             )
+            raise ConnectionRefusedError(error.args)
```
Comment (thewhaleking) [inline bittensor/core/subtensor.py:236]:
```suggestion
            raise
```

Then you don't have to worry about the `error` var.

**Review by thewhaleking — APPROVED** [review]
Two suggestions. If you use the second one, also remove the `as error` from line 228

File: bittensor/core/subtensor.py (hunk 1 above)
Comment (basfroman) [reply bittensor/core/subtensor.py:189]: There are reasons why I didn't do it

File: bittensor/core/subtensor.py [hunk 3]
```diff
@@ -223,14 +225,15 @@ def _get_substrate(self):
             except (AttributeError, TypeError, socket.error, OSError) as e:
                 logging.warning(f"Error setting timeout: {e}")
```
Comment (thewhaleking) [inline bittensor/core/subtensor.py:226]: This needs to be updated.

**Review by thewhaleking — CHANGES_REQUESTED** [review]

File: bittensor/core/subtensor.py (hunk 2 above)
Comment (basfroman) [reply bittensor/core/subtensor.py:236]: I want to show the user not only the stack, but also the code with the message

File: bittensor/core/subtensor.py (hunk 3 above)
Comment (basfroman) [reply bittensor/core/subtensor.py:226]: don't think so

**Review by thewhaleking — APPROVED** [review]

**Pull request merged**